task5/src/main/java/com/clouway/adapter/http/SecurityFilter.java

```diff
@@ -14,37 +15,33 @@
  */
 public class SecurityFilter implements Filter {
 
-  private UserRepository userRepository;
+  private SessionRepository sessionRepository;
+  private CookieFinder cookieFinder;
 
-  public SecurityFilter(UserRepository userRepository) {
-    this.userRepository = userRepository;
+  public SecurityFilter(SessionRepository sessionRepository, CookieFinder cookieFinder) {
+    this.sessionRepository = sessionRepository;
+    this.cookieFinder = cookieFinder;
   }
 
   public void init(FilterConfig filterConfig) throws ServletException {
-
   }
 
   public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
 
     HttpServletRequest request = (HttpServletRequest) servletRequest;
     HttpServletResponse response = (HttpServletResponse) servletResponse;
     Cookie[] cookies = request.getCookies();
-    String email = "";
-    for (int i = 0; i < cookies.length; i++) {
-      if (cookies[i].getName().equals("sessionId")) {
-        email = cookies[i].getValue();
-      }
-    }
-    User user = userRepository.findByEmail(email);
-    if (user != null) {
+    Cookie cookie = cookieFinder.find(cookies);
+    String sessionID = cookie.getValue();
```

Comment:
Here you need the sessionId so I would use `SidFinder` or `SidFetcher` like:
```java
String sid = sidFinder.find(request);
```
Can save 2 3 lines.
This is just suggestion do not apply because runnin out of time.